### PR TITLE
Pathways via ETO

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: app/models/concerns/eto/pathways_answers.rb [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+        relevant_section = section_starts_with('scoring - only')
+        answer_from_section(relevant_section, 'pathways assessment score')&.to_i
+      end
+
+      def rrh_desired_answer
+        # 2/17/2020
+        # 9A
+        return false unless hmis_assessment.pathways?
+
+        relevant_section = section_starts_with('Why do people take RRH')
```
Comment (awisema) [inline app/models/concerns/eto/pathways_answers.rb]: Are these case sensitive? That it was all lowercase in `assessment_score_answer` has me confused...

File: app/models/concerns/eto/pathways_answers.rb [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        # 2/17/2020
+        # 6C
+        return false unless hmis_assessment.pathways?
+
+        relevant_section = section_starts_with('Key Points to Share with the Participant Regarding Unit Preferences')
+        answer_from_section(relevant_section, '6c.')&.to_i
+      end
+
+      def required_minimum_occupancy_answer
+        # 2/17/2020
```
Comment (awisema) [inline app/models/concerns/eto/pathways_answers.rb]: Missing body? Maybe  a `# FIXME` like in `rrh_assessment_contact_info_answer`?

File: app/models/concerns/eto/pathways_answers.rb [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+      def requires_wheelchair_accessibility_answer
+        # 2/17/2020
+        # 6D
+        return false unless hmis_assessment.pathways?
+
+        relevant_section = section_starts_with('Key Points to Share with the Participant Regarding Unit Preferences')
+        answer_from_section(relevant_section, '6d.')&.downcase&.include?('wheelchair') || false
+      end
+
+      def requires_elevator_access_answer
```
Comment (awisema) [inline app/models/concerns/eto/pathways_answers.rb]: Both this and above are `6D`?

File: app/models/concerns/eto/pathways_answers.rb [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+      def youth_rrh_aggregate_answer
+        # 2/17/2020
+        # 9C
+        return nil unless hmis_assessment.pathways?
+
+        relevant_section = section_starts_with('Why do people take RRH instead of waiting for a subsidized housing unit like a Section 8 voucher?')
+        answer = answer_from_section(relevant_section, '9c.')
+        return nil unless answer
+
+        case answer.downcase
```
Comment (awisema) [inline app/models/concerns/eto/pathways_answers.rb]: I don't love this construction -- why not just an `if`?

File: app/models/concerns/eto/pathways_answers.rb [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+        when answer.starts_with?('Domestic Violence')
+          'dv'
+        when answer.starts_with?('Non-DV Programs Only')
+          'non-dv'
+        when answer.starts_with?('Both DV')
+          'both'
+        end
+      end
+
+      # def veteran_rrh_desired_answer
```
Comment (awisema) [inline app/models/concerns/eto/pathways_answers.rb]: Remove?

File: app/models/concerns/eto/pathways_answers.rb [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+      def neighborhood_interests_answer
+        # 2/17/2020
+        # 6H
+        return false unless hmis_assessment.pathways?
+
+        relevant_section = section_starts_with('Key Points to Share with the Participant Regarding Unit Preferences')
+        answer_from_section(relevant_section, '6h.')&.split('|')&.map(&:presence)&.compact
+      end
+
+      # def pathways_dv_score_answer
```
Comment (awisema) [inline app/models/concerns/eto/pathways_answers.rb]: Remove?

File: app/models/grda_warehouse/hmis_form.rb [hunk 7]
```diff
@@ -500,6 +559,29 @@ def <=>(other)
...
+      return false unless relevant_section.present?
+
+      relevant_section
+    end
+
+    #
+    # Finds the first relevant answer where the question includes the string.
+    #
+    # @param [Hash] section part of the HmisForm answers column
+    # @param [string>] question_string used to identify a relevant question
```
Comment (awisema) [inline app/models/grda_warehouse/hmis_form.rb]: `[String]`?

File: app/models/grda_warehouse/hmis_form.rb [hunk 8]
```diff
@@ -500,6 +559,29 @@ def <=>(other)
...
+      end&.first
+      return false unless relevant_section.present?
+
+      relevant_section
+    end
+
+    #
+    # Finds the first relevant answer where the question includes the string.
+    #
+    # @param [Hash] section part of the HmisForm answers column
```
Comment (awisema) [inline app/models/grda_warehouse/hmis_form.rb]: I like to include the parameter names, but that may just be my JavaDoc showing...

File: app/models/grda_warehouse/hud/client.rb [hunk 9]
```diff
@@ -639,7 +639,7 @@ def self.exists_with_inner_clients(inner_scope)
 
         if user.can_view_clients_with_roi_in_own_coc?
           # At a high level if you can see clients with ROI in your COC, you need to be able
-            # to see everyone for searching purposes.
+            # to see everyone for searching contact_info_for_rrh_assessmentrposes.
```
Comment (awisema) [inline app/models/grda_warehouse/hud/client.rb]: Accidental paste.

File: app/models/concerns/eto/pathways_answers.rb (hunk 1 above)
Comment (eanders) [reply app/models/concerns/eto/pathways_answers.rb]: I standardized this and moved the down casing to HmisForm.  So, case insensitive now.

File: app/models/concerns/eto/pathways_answers.rb (hunk 3 above)
Comment (eanders) [reply app/models/concerns/eto/pathways_answers.rb]: Yes, the question on the TouchPoint comes back pipe delimited.

**Pull request merged**